module/VuFindTheme/src/VuFindTheme/View/Helper/ConcatTrait.php

```diff
@@ -0,0 +1,324 @@
+    {
+        return $this->themeInfo->getBaseDir() . '/../local/cache/public/';
+    }
+
+    /**
+     * Using the concatKey, return the path of the concatenated file.
+     * Generate if it does not yet exist.
+     *
+     * @param array $group Object containing 'key' and stdobj file 'items'
+     *
+     * @return string
+     */
+    protected function getConcatenatedFilePath($group)
+    {
+        $urlHelper = $this->getView()->plugin('url');
+
+        // Don't recompress individual files
+        if (count($group['items']) === 1) {
+            $path = $this->getResourceFilePath($group['items'][0]);
+            $details = $this->themeInfo->findContainingTheme(
+                $this->getFileType() . '/' . $path,
+                ThemeInfo::RETURN_ALL_DETAILS
+            );
+            return $urlHelper('home') . 'themes/' . $details['theme']
+                . '/' . $this->getFileType() . '/' . $path;
+        }
+        // Locate/create concatenated asset file
+        $filename = md5($group['key']) . '.min.' . $this->getFileType();
+        $concatPath = $this->getResourceCacheDir() . $filename;
+        if (!file_exists($concatPath)) {
+            $minifier = $this->getMinifier();
+            foreach ($group['items'] as $item) {
+                $details = $this->themeInfo->findContainingTheme(
+                    $this->getFileType() . '/' . $this->getResourceFilePath($item),
+                    ThemeInfo::RETURN_ALL_DETAILS
+                );
+                $minifier->add($details['path']);
+            }
+            $minifier->minify($concatPath);
+        }
+
+        return $urlHelper('home') . 'cache/' . $filename;
+    }
+
+    /**
+     * Process and return items in index order
+     *
+     * @param string|int $indent Amount of whitespace/string to use for indention
+     *
+     * @return string
+     */
+    protected function outputInOrder($indent)
+    {
+        // Some of this logic was copied from HeadScript; it does not all apply
+        // when incorporated into HeadLink, but it has no harmful side effects.
+        $indent = (null !== $indent)
+            ? $this->getWhitespace($indent)
+            : $this->getIndent();
+
+        if ($this->view) {
+            $useCdata = $this->view->plugin('doctype')->isXhtml();
+        } else {
+            $useCdata = $this->useCdata;
+        }
+
+        $escapeStart = ($useCdata) ? '//<![CDATA[' : '//<!--';
+        $escapeEnd   = ($useCdata) ? '//]]>' : '//-->';
+
+        $output = [];
+        foreach ($this->groups as $group) {
+            if (isset($group['other'])) {
+                $output[] = $this->itemToString(
+                    $group['item'], $indent, $escapeStart, $escapeEnd
+                );
+            } else {
+                $path = $this->getConcatenatedFilePath($group);
+                $item = $this->setResourceFilePath($group['items'][0], $path);
+                $output[] = parent::itemToString(
```

Comment:
Why are we using parent::itemToString here and $this->itemToString above in line 265? If this is a meaningful distinction, we should comment it. If not, we should switch to something more consistent.